backend/src/services/token/token.service.ts

```diff
@@ -29,6 +36,11 @@ class Token {
 
     return jwtVerify(token, secretKey);
   }
-}
 
+  async decode<T>(token: string): Promise<T> {
+    const data = decodeJwt(token);
+
+    return data as JWTPayload & T;
+  }
```

Comment:
```suggestion
  async decode<T>(token: string): Promise<JWTPayload & T> {
    const data = decodeJwt(token);

    return data as JWTPayload & T;
  }
```
keep things consistent